### DISCUSSION TITLE
Problem with touchscreen on RVT70HSBNWC00-B

### DISCUSSION THREAD

**Discussion by ADTL** [RudolphRiedel/FT800-FT813#57] [General]
Hi, 

I have two displays RVT70HSBNWC00-B. 
Display working fine, but touchscreen not working in both LCD's.
Was tried on Arduino UNO, ESP32, STM32F103 with same results - display working, touchscreen not.
P.S. With display RVT70UGFNWC01 (FT813) all working fine....
just now test this libs with RVT50HQBNWC00 they have near the same touchscreen (seems the same like in RVT70HSBNWC00-B)
and this display (RVT50HQBNWC00) working with touchscreen fine.
Can you advice something?

With regards, 
Roman.

**Comment by RudolphRiedel** [comment]
I do have a RVT70HSBNWC00-B and touch is running just fine (at least single-touch, multi-touch is a different story but still works to some degree).
As you tested it with an UNO that rules out that the SPI clock is too high, with my hardware touch is generally not working above 12MHz to 16MHz.
And since the RVT50HQBNWC00 is practically identicle in this regard this points  to one thing left I encountered.

Have you made sure that your supply voltages are correct? Especially the 3.3V digital supply needs to be stable as the touch controllers tend to be susceptible to short voltage dips in that these are not as forgiving as the EVE chips.
Check that you do not have a voltage drop when you switch on the backlight for example.

I designed my own controller board for the RVT101H as it draws quite some power but thankfully allows the backlight
to be powered with a higher voltage than the usual 5V.
I am using this board for the RVT70H as well with 5V for the backlight.
My board has two switching regulators that can deliver 1.2A and operate at 1.4MHz.
The first is exclusively used for the backlight.
The second is set to 3.36V (by design a bit over 3.3V) and powers the logic on my board and is connected
with an extra 4.7µF placed directly at the FFC.

My setup with the RVT70H draws 226mA from 10V right now which means that the backlight that is tuned down to only 0x30 currently draws >350mA.

**Reply by ADTL** [reply]
Thanks a lot for the fast and detailed answer.
At first I tried to power the display from demo boards, and with all versions of LCD, except the new ones RVT70HSBNWC00-B, it worked fine. 
After your advice, I used a laboratory power supply that has a ripple of less than 30 mV at the output. 
Unfortunately, the result remained the same - the touchscreen does not work.
I wrote to Riverdi support with the serial numbers of the displays, as a result I received a RMA number and I will send the displays back to them. We'll see how it ends.

With regards,
Roman.

**Reply by RudolphRiedel** [reply]
That is odd.

**Comment by MaWa53** [comment]
Did you also think of the 3.3V power supply? 
I first connected both the 5V and the 3.3V to a Raspi 4. 
Although the screen worked the touch function had occasional glitches. 
When I lowered the backlight value the glitches were less but still existed. So I supposed the power supply was the cause.

As the project I am working on contains a Raspi 4, a RVT70HSBNWC00-B plus a Bigtreetech Octopus Controller board (which has a much higher current capacity than the Raspi4), I connected the displays 5V and 3.3V plus GND to the Octopus and only GND plus SPI plus CS and PD to the Raspberry 4. 
Voilá Touch - even multiple touch - works fine.
